lua/autorun/server/cfc_attention_monitoring_init.lua

```diff
@@ -1,22 +1,70 @@
-util.AddNetworkString( "CFC_AttentionMonitor_GameHasFocus" )
+util.AddNetworkString( "CFC_AttentionMonitor" )
 
-CFCAttentionMonitor = { pendingFocusChanges = {} }
-local pendingFocusChanges = CFCAttentionMonitor.pendingFocusChanges
+CFCAttentionMonitor = CFCAttentionMonitor or {}
+CFCAttentionMonitor.PlayerStatuses = {}
 
 hook.Add( "PlayerDisconnected", "CFC_AttentionMonitor_CleanupPlayerData", function( ply )
-    pendingFocusChanges[ply] = nil
+    CFCAttentionMonitor.PlayerStatuses[ply] = nil
 end )
 
-local function focusCallback( _, ply )
-    pendingFocusChanges[ply] = not net.ReadBool()
+local function sync( ply )
+    local plyTable = CFCAttentionMonitor.PlayerStatuses[ply]
+    local hasStatus = false
+    for i = 1, CFCAttentionMonitor.EnumCount do
+        if plyTable[i] then
+            ply:SetNW2Int( "CFC_AM_Type", i )
+            ply:SetNW2Int( "CFC_AM_Time", CurTime() )
+            hasStatus = true
+
+            break
+        end
+    end
+
+    if not hasStatus then
+        ply:SetNW2Int( "CFC_AM_Type", 0 )
+    end
+end
+
+-- Leaky bucket rate limiting
+local buffer = 50 -- total in the bucket
+local refill = 8 -- per second
+local function rateLimit( ply )
```

Comment:
I would suggest separating the sync process from the net receiver. If you sync on a timer, and use the net message to update a "thisIsMyCurrentStatus" variable.

Then the sync can actually network the sync, at an expected interval.

No rate limiting needed, still no excess networking (SetNW2 only sends changes), and still protected from spam attempts